src/test/suite/providers/formatting/formattingProvider.utils.test.ts

```diff
@@ -655,6 +659,278 @@ suite('FormattingProvider utils', () => {
+                rs: [-1],
+            },
+        ];
+        dataList.forEach((data) => {
+            test('[' + data.in.depth + '] => [' + data.rs + ']', () => {
+                assert.deepStrictEqual(
+                    data.in.exitBlockOfCode(data.bn),
+                    data.rs,
+                );
+            });
+        });
+    });
+
+    suite('FlowOfControlNestDepth.pop', () => {
+        // List of test data
+        let dataList = [
+            // {
+            //     in: , // input array
+            //     rs: , // expected result
+            // },
+            {
+                in: new FlowOfControlNestDepth([-1]),
+                rs: [-1],
+            },
+        ];
+        dataList.forEach((data) => {
+            test('[' + data.in.depth + '] => [' + data.rs + ']', () => {
+                data.in.pop();
+                assert.deepStrictEqual(data.in.depth, data.rs);
+            });
+        });
+    });
+
+    suite('FlowOfControlNestDepth.restoreEmptyDepth', () => {
+        // List of test data
+        let dataList = [
+            // {
+            //     in: , // input array
+            //     rs: , // expected result
+            // },
+            {
+                in: new FlowOfControlNestDepth([]),
+                rs: [-1],
+            },
+        ];
+        dataList.forEach((data) => {
+            test('[' + data.in.depth + '] => [' + data.rs + ']', () => {
+                data.in.restoreEmptyDepth();
+                assert.deepStrictEqual(data.in.depth, data.rs);
+            });
+        });
+    });
+
+    suite('FlowOfControlNestDepth.restoreDepth', () => {
+        // List of test data
+        let dataList = [
+            // {
+            //     in: , // input array
+            //     rs: , // expected result
+            //     dp: , // depth
+            // },
+            {
+                in: new FlowOfControlNestDepth([-1, 0, -1, 1, 2]),
+                rs: 1,
+                dp: [-1, 0, -1],
+            },
+            {
+                in: new FlowOfControlNestDepth([-1]),
+                rs: undefined,
+                dp: [-1],
+            },
+        ];
+        dataList.forEach((data) => {
+            test('[' + data.in.depth + "] => '" + data.rs + "'", () => {
+                assert.strictEqual(data.in.restoreDepth(), data.rs);
+            });
+            test('[' + data.in.depth + '] => [' + data.dp + ']', () => {
+                data.in.restoreDepth();
+                assert.deepStrictEqual(data.in.depth, data.dp);
+            });
+        });
+    });
+
+    suite('alignSingleLineComments', () => {
+        // List of test data
+        let dataList = [
+            // {
+            //     in: , // input test string
+            //     op: , // formatting options
+            //     rs: , // expected result
+            // },
+            {
+                in: '',
+                op: { insertSpaces: true, tabSize: 4, preserveIndent: true },
```

Comment:
For `op`, let's use the `defaultArgs` pattern introduced with this code. https://github.com/mark-wiemer/vscode-autohotkey-plus-plus/blob/29403f32a54d9526e1d824f0876372f6a0dbceaa/src/test/suite/service/runnerService.test.ts#L7

Reply:
I not understand, why you wrote twice same parameters in declaration `defaultArgs`:

```TypeScript
const defaultArgs: {
    compilePath: 'mockCompilePath';
} = {
    compilePath: 'mockCompilePath',
}
```

Why not like this?

```TypeScript
const defaultArgs: {
    compilePath: string;
} = {
    compilePath: 'mockCompilePath',
}
```

Reply:
Done with `defaultFormattingOptions`.